app/internal/security/ouath2.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime, timedelta
+from typing import Union
+
+from app.config import JWT_ALGORITHM, JWT_KEY, JWT_MIN_EXP
+from passlib.context import CryptContext
+from app.database.models import User
+from fastapi import Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from jwt.exceptions import InvalidSignatureError
+from sqlalchemy.orm import Session
+from starlette.requests import Request
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_401_UNAUTHORIZED
+from .schema import LoginUser, CurrentUser
+
+
+pwd_context = CryptContext(schemes=["bcrypt"])
+oauth_schema = OAuth2PasswordBearer(tokenUrl="/login")
+
+
+async def get_db_user_by_username(db: Session, username: str) -> User:
+    """Checking database for user by username field"""
+    user = db.query(User).filter_by(username=username).first()
+    return user
+
+
+def get_hashed_password(password: str) -> str:
+    """Hashing user password"""
+    return pwd_context.hash(password)
+
+
+def verify_password(plain_password: str, hashed_password: str) -> bool:
+    """Verifying password and hashed password are equal"""
+    return pwd_context.verify(plain_password, hashed_password)
```

Comment:
Do we have a salt for each user?

Reply:
https://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt.html
auto generated by the 'bcrypt' scheme i use. as well as 12 rounds

Reply:
Awesome!